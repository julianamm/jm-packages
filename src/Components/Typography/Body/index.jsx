import React from 'react';
import PropTypes from 'prop-types';
import StyledBody from './styles';

const Body = props => <StyledBody {...props} >{props.label}</StyledBody>

Body.defaultProps = {
    xs: false,
    sm: false,
    md: false,
    lg: false,
    color: 'Blue-Grey-800',
    textCenter: false,
    hover: false,
    cursor: false,
}

Body.propTypes = {
    label: PropTypes.string.isRequired,
    color: PropTypes.string,
    xs: PropTypes.bool,
    sm: PropTypes.bool,
    md: PropTypes.bool,
    lg: PropTypes.bool,
    textCenter: PropTypes.bool,
    hover: PropTypes.bool,
    cursor: PropTypes.bool,
}

export default Body;