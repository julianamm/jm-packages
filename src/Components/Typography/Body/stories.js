import React from 'react'
import { storiesOf } from '@storybook/react'
import Body from '.'

storiesOf('Typography|Body', module)

.add('Default', () => (
    <div>
        <div>
            <Body label={"Body component default"} />
        </div>
        <div>
            <Body xs label={"Body component XS"} />
        </div>
        <div>
            <Body sm label={"Body component SM"} />
        </div>
        <div>
            <Body md label={"Body component MD"} />
        </div>
        <div>
            <Body lg label={"Body component LG"} />
        </div>
    </div>
))

.add('XS', () => (
  <Body xs label={"Body component"} />
))

.add('SM', () => (
  <Body sm label={"Body component"} />
))

.add('MD', () => (
  <Body md label={"Body component"} />
))

.add('LG', () => (
  <Body lg label={"Body component"} />
))

.add('Colorful', () => (
  <div>
      <p><Body color={"Blue-600"} lg label={"Body component Blue-600"} /></p>
      <p><Body color={"Pink-400"} xs label={"Body component Pink-400"} /></p>
      <p><Body color={"Indigo-900"} lg label={"Body component Indigo-900"} /></p>
      <p><Body color={"Light-Green-A400"} md label={"Body component Light-Green-A400"} /></p>
      <p><Body color={"Brown-300"} label={"Body component Brown-300"} /></p>
      <p><Body color={"Red-700"} lg label={"Body component Red-700"} /></p>
  </div>
))

