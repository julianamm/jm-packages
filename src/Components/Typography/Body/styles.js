import styled from 'styled-components'
import Color from '../../Colors'

export default styled.span`
    font-size: 14px;
    line-height: 18px;

    ${({ color }) => color && `
        color: ${Color[color]};
    `}

    ${({ xs }) => xs && `
        font-size: 12px;
        line-height: 16px;
    `}

    ${({ sm }) => sm && `
        font-size: 14px;
        line-height: 18px;
    `}

    ${({ md }) => md && `
        font-size: 16px;
        line-height: 20px;
    `}

    ${({ lg }) => lg && `
        font-size: 18px;
        line-height: 22px;
    `}

    ${({ textCenter }) => textCenter && `
    text-align: center;
    `}
    
    ${({ hover }) => hover && `
        :hover {
            text-decoration: underline;
        }
    `}

    ${({ cursor }) => cursor && `
        cursor: pointer;
    `}

`