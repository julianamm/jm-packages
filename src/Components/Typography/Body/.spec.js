import React from 'react'
import renderer from 'react-test-renderer'

import Body from '.'

describe('Typography', () => {
    it('Rendering component without crashing', () => {
        const tree = renderer.create(
            <Body
                color={"Indigo-900"}
                lg
                label={"Body component Indigo-900"}
            />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})