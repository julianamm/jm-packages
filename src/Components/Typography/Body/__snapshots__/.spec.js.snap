// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Rendering Font Components without crashing 1`] = `
.c0 {
  font-size: 14px;
  line-height: 18px;
  color: #1A237E;
  font-size: 18px;
  line-height: 22px;
}

<span
  className="c0"
  color="Indigo-900"
  label="Body component Indigo-900"
>
  Body component Indigo-900
</span>
`;
